api_app/api/routes/workspaces.py

```diff
@@ -101,7 +104,7 @@ async def retrieve_workspace_by_workspace_id(workspace=Depends(get_workspace_by_
 
 
 @workspaces_core_router.post("/workspaces", status_code=status.HTTP_202_ACCEPTED, response_model=WorkspaceIdInResponse, name=strings.API_CREATE_WORKSPACE, dependencies=[Depends(get_current_admin_user)])
-async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo=Depends(get_repository(WorkspaceRepository))) -> WorkspaceIdInResponse:
+async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo=Depends(get_repository(WorkspaceRepository)), operations_repo=Depends(get_repository(OperationRepository))) -> WorkspaceIdInResponse:
```

Comment:
I suggest returning the operation id instead of workspace id (also when doing updates).

When modifying a workspace, workspace service or a user ressource, the initiator wants to know what is the status of that change. That info can be retrieved via the operation. When the operation is done, then then workspace id becomes relevant.
If the workspace id is needed up front, then lookup the operation first, which will contain the workspace id

Reply:
Yes i've been wondering this, and if we're happy with this model I think it makes sense. Do we take it a step towards a HATEOAS structure and return something like: 

```json
HTTP STATUS 202 (Accepted)
{
	"operation": {
		"href": "/api/workspaces/<workspace-guid-here>/operations/<op-guid-here>",
		"operation_id": "guid-here"
                "resource_id": "resource-guid-here"
	}
}
```

?

Reply:
@marrobi what do you think to this?

Reply:
It would be good to get both the operation ID and resource ID back. For example if want to get the calculated properties of a new resource immediately after create.

Although in the first instance operation id might be sufficient.

Reply:
Here's what i've gone for - think it covers all the requirements. `resourcePath` was added to the resource object and to the operation as I need to know the path so I can query the right endpoint for the ops. Also added a `Location` header to the `POST` and `DELETE` endpoints which will refer you to the correct ops endpoint to poll.

```json
{
  "operation": {
    "id": "fc2e867d-c726-424d-8f64-13392b23bbef",
    "resourceId": "b374eb9b-7fd1-4448-8af5-7e02ef2749ee",
    "resourcePath": "/workspaces/a1bceefb-092a-4163-b67a-2a40e49d3ea3/workspace-services/343d3543-33cb-4cc8-9e35-1e38a0e0c6a5/user-resources/b374eb9b-7fd1-4448-8af5-7e02ef2749ee",
    "resourceVersion": 0,
    "status": "not_deployed",
    "message": "This resource has not yet been deployed",
    "createdWhen": 1643718851.996725,
    "updatedWhen": 1643718851.996725
  }
}
```